raiden-dapp/src/components/AddressInput.vue

```diff
@@ -100,41 +117,186 @@ export default class AddressInput extends Mixins(BlockieMixin) {
     // as the input being invalid. Since the :rules prop does not support
     // async rules we have to workaround with a reactive prop
     const isAddressValid =
-      !this.busy && this.errorMessages.length === 0 && this.isAddressAvailable;
+      !this.busy &&
+      !this.typing &&
+      this.errorMessages.length === 0 &&
+      this.isAddressAvailable;
+
     if (isAddressValid) {
       this.input(this.address);
     }
 
     return [() => isAddressValid || ''];
   }
 
+  private checkAvailability(
+    value: ValidationResult
+  ): Observable<ValidationResult> {
+    return iif(
+      () => 'error' in value,
+      of(value),
+      of(value).pipe(
+        tap(() => (this.busy = true)),
+        switchMap(value => {
+          return iif(
+            () => value.value in this.presences,
+            of(this.presences[value.value]),
+            defer(() => from(this.$raiden.getAvailability(value.value)))
+          );
+        }),
+        tap(available => {
```

Comment:
`tap` and `map` below would run serially and would consist of only synchronous operators, I'd say there's no need for them to be split, you can just put the `tap` statements in the beginning of the `map` callback